### ISSUE TITLE
Add Bash install for macOS

### ISSUE THREAD

**Issue by conradwt** [Azure/azure-cli#7362]
**Is your feature request related to a problem? Please describe.**

At this time, the only way to install the Azure CLI is Homebrew if one is reading the documentation.  However, I really don't want to install yet another package manager but rather use a script.

**Describe the solution you'd like**

I would like to be able to install Azure CLI using Bash script.

**Describe alternatives you've considered**

I have tried using pip during a Cloud Native workshop which failed.  Also, I have used Docker with success but it would be nice not to force this dependency on users of the Azure CLI.

**Additional context**

In short, I would like to be able to install Azure CLI without requiring the use of any package manager which would be great for workshops and other places where you would like to get things working.

**Comment by yugangw-msft** [comment]
//CC @troydai 
@conradwt, what error you got when used `pip`?
The other option is to git clone CLI's repository and then run [dev_setup](https://github.com/Azure/azure-cli/blob/dev/scripts/dev_setup.py)

**Comment by troydai** [comment]
@conradwt does PIP work you?

**Comment by conradwt** [comment]
I was in a Cloud Native workshop at GopherCon 2018 and `pip` didn’t work for me. I received an error/warring saying that it couldn’t load the search module.  I’ll post a formal bug report.  Next, `pip` needs to be documented as a way to install Azure CLI for macOS.  At this time, I’m only seeing Homebrew.

**Comment by troydai** [comment]
Did you try this: https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?view=azure-cli-latest#install-or-update?

**Comment by conradwt** [comment]
I didn’t try this because the documentation says that it’s for a Linux install of the Azure CLI.

**Comment by troydai** [comment]
Please give it a try.

**Comment by troydai** [comment]
Also, please make sure the PIP works correctly on your machine. Even with the script, the installation is centralized around pip.

**Comment by conradwt** [comment]
@troydai Pip works just fine on my system because I have used it with several different projects with error.
